test/exercises/Mandelbrot/solutions/mandelbrot4-taskpar.chpl

```diff
@@ -17,7 +17,7 @@ config const maxSteps = 50;
 // 
 // The number of tasks we should spawn.
 //
-config const numTasks = here.numCores; // Cores on locale #0.
+config const numTasks = here.numPUs(physical=true); // Cores on locale #0.
```

Comment:
I believe these are really two different things, or maybe even more than that. MAX_LOGICAL and MAX_PHYSICAL are settings for CHPL_RT_NUM_THREADS_PER_LOCALE, which tell the tasking layer how many threads should be used to execute tasks. The old numCores var, and the numPUs() proc proposed to replace it, would be used to determine the actual value of the number of logical or physical processors corresponding to those symbolic values.

The code here is setting an application-level value which will be the number of tasks it's going to create in a coforall intended (I think, though it isn't documented) to use all of the processor resources available, for some definition of "available". I'm not sure that numCores was the right thing here in the first place. I think perhaps here.maxTaskPar might be best. Possibly application code might want to query hardware characteristics to set the number of tasks, but having done so it certainly ought to compare that to maxTaskPar and say something if it finds itself using a tasking layer with different ideas about achieving performance, which presumably is the point.

Having said all this, I'll add that the change I made here was just to make sure that numTasks has the value described by its comment after my change. Note that this is different from the old value, since numCores, despite its name and the comment on the line of code, actually was the number of logical processors (hardware threads), not physical ones (cores).